scripts/tidycensus.py

```diff
@@ -0,0 +1,220 @@
+                '''.format(path=path, dataset=dataset_name)
+
+                ro.r(r_string)
+                file_paths[dataset_name] = os.path.normpath(
+                    os.path.join(path, '{}.shp'.format(dataset_name)))
+
+            else:
+                r_string = '''
+                library(tidycensus)
+                rdf = as.data.frame.data.frame({})
+                '''.format(dataset_name)
+
+                ro.r(r_string)
+                rdf = ro.r('rdf')
+
+                with localconverter(ro.default_converter + pandas2ri.converter):
+                    pdf = ro.conversion.rpy2py(rdf)
+                path = os.path.normpath(
+                    self.engine.format_filename('{}.csv'.format(dataset_name)))
+                pdf.to_csv(path, index=False)
+                file_paths[dataset_name] = path
+
+    def download(self, engine=None, debug=False):
+        Script.download(self, engine, debug)
+        engine = self.engine
+        datasets = [
+            'fips_codes', 'mig_recodes', 'pums_variables', 'state_laea', 'county_laea'
+        ]
+        key = ro.r('Sys.getenv("CENSUS_API_KEY")')
+
+        data_path = os.path.normpath(DATA_WRITE_PATH.format(dataset='tidycensus'))
+        if not os.path.exists(data_path):
+            engine.create_raw_data_dir()
+
+        download = True
+        filepaths = [
+            'fips_codes.csv', 'mig_recodes.csv', 'pums_variables.csv',
+            'state_laea/state_laea.shp', 'county_laea/county_laea.shp'
+        ]
+        for filepath in filepaths:
+            if os.path.isfile(os.path.join(data_path, filepath)):
+                filename, file_ext = os.path.splitext(os.path.basename(filepath))
+                file_paths[filename] = os.path.join(data_path, filepath)
+                download = False
+
+        if (key[0]
+                == '') and (len([x for x in packages if rpackages.isinstalled(x)]) > 0):
+            result = self.install_tidycensus()
+            print("Installing tidycensus...")
+            if result:
+                print("Installation completed!!!")
+                print(
+                    '*The API key provided to you from the Census formated in quotes. A key can be acquired at http://api.census.gov/data/key_signup.html*'
+                )
+                print("Enter your census api key here : ", end="")
+                api_key = input()
+                self.census_api_key(api_key)
+            else:
+                print("Installation failed!!!")
+                exit()
+            self.download_raw_data(datasets)
+
+        elif (key[0] == '') and (len([x for x in packages if rpackages.isinstalled(x)])
+                                 == 0):
+            print(
+                '*The API key provided to you from the Census formated in quotes. A key can be acquired at http://api.census.gov/data/key_signup.html*'
+            )
+            print("Enter your census api key here : ", end="")
+            api_key = input()
```

Comment:
We should not ask people to enter any information, they should use config files for keys